### PR TITLE
fix(feeds): preserve cadence and skip unchanged KEV updates

### LINKED ISSUE #4204: fix(advisory-feeds): stop reconciliation-cadence KEV rewrites
## Describe the bug

Advisory-feed jobs can run on the scheduler's reconciliation cadence instead of their configured refresh intervals. KEV feeds also rewrite unchanged corpora and emit a misleading `skip guard appears inert` error.

This was noticed while verifying #4183, but it is not caused by that change. The behavior is already present in `v1.4.49` and `staging`.

## Root cause

There are two related defects:

1. The worker attempts to schedule its successor before the currently executing Oban job returns. Worker uniqueness includes the `executing` state, so that insertion collides with or reuses the current job. Once the job completes, no future successor remains. The reconciliation scheduler then seeds another run on its next safety tick.
2. KEV records do not carry a usable modification timestamp. The loader correctly treats a missing timestamp as unknown rather than unchanged, but the generic inert-guard alarm interprets zero skipped records against any non-empty corpus as a broken guard. For KEV, that alarm is a false positive while the corpus is nevertheless rewritten.

## Expected behavior

- A completed feed run leaves exactly one future job scheduled at the configured refresh interval.
- The reconciliation tick repairs a genuinely missing schedule without becoming the normal execution cadence.
- Unchanged KEV entries do not cause a full database rewrite on every run.
- The inert-guard alarm only fires for feeds where unchanged records can be determined reliably.
- Timestamp-based NVD deduplication and its safety alarm remain intact.

## Acceptance criteria

- [ ] Add a regression test proving successor scheduling works while the current job is still `executing` and preserves the configured interval.
- [ ] Add a scheduler test proving reconciliation does not enqueue a duplicate or early feed run when a valid future job exists.
- [ ] Add KEV coverage proving an unchanged corpus is not fully rewritten on every scheduled run, using a stable comparison mechanism suitable for feeds without modification timestamps.
- [ ] Add guard coverage proving KEV does not emit a false inert-guard error while a timestamp-capable feed still alarms when its skip guard actually stops working.
- [ ] Verify feed job timestamps follow their configured intervals after rollout.
- [ ] Verify NVD deduplication does not regress to full-corpus rewrites.

## Impact

The defect creates avoidable database/WAL load and continuous error-log noise. Rolling back #4183 does not mitigate it because the affected scheduling and feed-loading behavior predates that change.

### PR DESCRIPTION
**Pull request by mfreeman451**
Closes #4204.

## Summary

- schedule advisory-feed successors with scheduled-only uniqueness so a worker can enqueue its next run while the current job is still executing
- retain the worker's existing incomplete-state uniqueness for seeding and reconciliation
- persist deterministic SHA-256 content hashes for `cisa-kev` and `vulncheck-kev`, including canonical advisory and coordinate data, so unchanged KEV payloads do not rewrite rows
- treat legacy KEV rows with a null hash as changed once, allowing the first post-upgrade run to backfill the hash
- keep `modified_at` comparison for non-KEV feeds and count only usable feed-specific comparison state in the inert guard
- add the nullable column through a post-baseline migration while preserving the frozen baseline boundary
- synchronize the Helm migration guard and integration-test projections with the new migration and regressions

## Verification

- `make test`: 208/208 tests passed on corrected head `d9163830f1` ([BuildBuddy invocation](https://carverauto.buildbuddy.io/invocation/9b05d837-ed82-4052-92ad-28d097a667fb)); one unrelated notifications shard passed on Bazel's automatic second attempt
- exact baseline metadata and post-baseline migration startup-safety checks passed
- focused scratch-database scheduler and loader regressions: 11 tests, 0 failures
- focused scheduler uniqueness, loader, migration/schema, and CI/Helm contract gates passed
- final whole-branch and CI-correction reviews found no outstanding Critical, Important, or Minor findings

## Rollout note

The post-baseline migration adds a nullable `content_hash`. Existing KEV rows therefore rewrite once on their first successful post-upgrade feed run to populate the hash; later unchanged runs are skipped. Other advisory feeds retain their existing timestamp comparison behavior.

### CODE REVIEW

File: elixir/serviceradar_core/lib/serviceradar/inventory/advisory_feeds/loader.ex [hunk 1]
```diff
@@ -336,8 +432,12 @@ defmodule ServiceRadar.Inventory.AdvisoryFeeds.Loader do
...
       chunk
-      |> Enum.map(fn %{advisory: advisory} ->
-        advisory_row(advisory, provider, feed_key, generation, now)
+      |> Enum.map(fn %{advisory: advisory} = record ->
+        record
+        |> content_hash()
+        |> then(
+          &Map.put(advisory_row(advisory, provider, feed_key, generation, now), :content_hash, &1)
+        )
       end)
```
[the remark this answers is not in the document]
Comment (mfreeman451) [reply elixir/serviceradar_core/lib/serviceradar/inventory/advisory_feeds/loader.ex:443]: Addressed in 96030e7f64. The loader now selects comparison mode once per chunk and calls content_hash/1 only for :content_hash feeds (cisa-kev and vulncheck-kev); timestamp-comparison feeds persist nil. I added a Postgres integration regression test that failed against the previous implementation and now passes. Verification: the full loader integration file passed (10 tests), formatting passed, and make test passed all 208/208 targets.

File: elixir/serviceradar_core/test/serviceradar/inventory/advisory_feeds/feed_worker_scheduling_test.exs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+          where: fragment("?->>'feed' = ?", job.args, "nist-nvd2"),
+          where: job.id != ^executing.id and job.state == "scheduled"
+        )
+      )
+
+    assert [%Oban.Job{conflict?: false, args: %{"feed" => "nist-nvd2"}} = successor] =
+             successors
+
+    delay = DateTime.diff(successor.scheduled_at, DateTime.utc_now())
+    assert delay in 7_195..7_200
```
[the remark this answers is not in the document]
Comment (mfreeman451) [reply elixir/serviceradar_core/test/serviceradar/inventory/advisory_feeds/feed_worker_scheduling_test.exs]: Addressed in 96030e7f64. The test now brackets FeedWorker.perform/1 with before/after timestamps, subtracts the exact 7,200-second cadence from the persisted scheduled_at value, and verifies that scheduling origin falls inside that bracket. This excludes post-perform query latency without weakening the exact cadence assertion. Verification: the focused scheduling file passed (2 tests), formatting passed, and make test passed all 208/208 targets.
